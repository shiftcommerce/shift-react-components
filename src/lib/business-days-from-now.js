// Libraries
import addDays from 'date-fns/add_days'
import isWeekend from 'date-fns/is_weekend'

export default function businessDaysFromNow (days) {
  // Convert number of days to integer
  const parsedDays = parseInt(days)

  // Calculate start date
  const startDate = new Date(Date.now())

  // Calculate end date (startDate + parsedDays)
  let endDate = addDays(startDate, parsedDays)

  // Iterate over endDate, increasing by 1 day until it is a business day (Mon - Fri)
  while (isWeekend(endDate)) {
    endDate = addDays(endDate, 1)
  }

  return endDate
}
