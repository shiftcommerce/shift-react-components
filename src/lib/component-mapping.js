import Config from './config'

/**
 * Lib
 */
import FormErrors from './form-errors'

/**
 * Navigation Components
 */
import NavBar from '../components/navigation/navbar'
import NavBarOption from '../components/navigation/navbar-option'

/**
 * Objects
 */
import Breadcrumb from '../objects/breadcrumb'
import Button from '../objects/button'
import Checkbox from '../objects/checkbox'
import ConditionalLink from '../objects/conditional-link'
import DropdownSelect from '../objects/dropdown-select'
import Head from '../objects/head'
import Image from '../objects/image'
import Input from '../objects/input'
import LazyLoad from '../objects/lazy-load'
import Link from '../objects/link'
import Loading from '../objects/loading'
import Logo from '../objects/logo'
import PayPalButton from '../objects/paypal-button'
import Rating from '../objects/rating'
import VariantSelector from '../objects/variant-selector'

/**
 * PDP Components
 */
import ProductCarousel from '../components/products/display/product-carousel'
import ProductEwisForm from '../components/products/display/product-ewis-form'
import ProductPrice from '../components/products/display/product-price'

/**
 * PLP Components
 */
import ProductListingCard from '../components/products/listing/product-listing-card'
import ProductMenu from '../components/products/listing/product-menu'
import ProductMenuOptions from '../components/products/listing/product-menu-options'

/**
 * Cart/Checkout components
 */
import AddressBook from '../components/checkout/address-book'
import AddressForm from '../components/checkout/address-form'
import AddressFormHeader from '../components/checkout/address-form-header'
import LineItems from '../components/cart/line-items'
import PaymentIcons from '../components/cart/payment-icons'
import PaymentMethods from '../components/checkout/payment-methods'
import PaymentMethod from '../components/checkout/payment-method'
import PaymentMethodHeader from '../components/checkout/payment-method-header'
import ShippingMethodsHeader from '../components/checkout/shipping-methods-header'
import StripeFields from '../components/checkout/stripe-fields'
import StripePayment from '../components/checkout/stripe-payment'
import StripeWrapper from '../components/checkout/stripe-wrapper'

/**
 * Order Components
 */
import OrderLineItems from '../components/orders/order-line-items'
import OrderList from '../components/orders/order-list'
import ShippingAddresses from '../components/orders/shipping-addresses'

/**
 * Search Components
 */
import SearchBar from '../components/search/search-bar'
import SearchFilters from '../components/search/search-filters'
import SearchHits from '../components/search/search-hits'
import SearchRatingFilter from '../components/search/search-rating-filter'
import SearchRefinements from '../components/search/search-refinements'
import SearchSlider from '../components/search/search-slider'

/**
 * Static Page
 */
import StaticPageError from '../components/static-page/error'
import StaticPageErrorBody from '../components/static-page/error-body'
import StaticPageErrorDetails from '../components/static-page/error-details'

/**
 * Layout Components
 */
import CustomHead from '../components/layout/custom-head'
import Footer from '../components/layout/footer'
import Minibag from '../components/layout/minibag'

const mapping = {
  AddressBook,
  AddressForm,
  AddressFormHeader,
  Breadcrumb,
  Button,
  Checkbox,
  ConditionalLink,
  CustomHead,
  DropdownSelect,
  Footer,
  FormErrors,
  Head,
  Image,
  Input,
  LazyLoad,
  LineItems,
  Link,
  Loading,
  Logo,
  Minibag,
  NavBar,
  NavBarOption,
  OrderLineItems,
  OrderList,
  PayPalButton,
  PaymentIcons,
  PaymentMethods,
  PaymentMethod,
  PaymentMethodHeader,
  ProductCarousel,
  ProductEwisForm,
  ProductListingCard,
  ProductMenu,
  ProductMenuOptions,
  ProductPrice,
  Rating,
  SearchBar,
  SearchFilters,
  SearchHits,
  SearchRatingFilter,
  SearchRefinements,
  SearchSlider,
  ShippingAddresses,
  ShippingMethodsHeader,
  StaticPageError,
  StaticPageErrorBody,
  StaticPageErrorDetails,
  StripeFields,
  StripePayment,
  StripeWrapper,
  VariantSelector
}

export default function (module) {
  return Config.get()[module] || mapping[module]
}
