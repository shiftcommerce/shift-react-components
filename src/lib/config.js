class Config {
  constructor () {
    this.config = {}
  }

  set (newConfig) {
    this.config = Object.assign(this.config, newConfig)
    return this.config
  }

  get () {
    return this.config
  }

  reset () {
    this.config = {}
  }
}

export default new Config()
