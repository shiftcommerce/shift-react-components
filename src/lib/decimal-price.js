export function decimalPrice (price) {
  return (price).toFixed(2)
}
