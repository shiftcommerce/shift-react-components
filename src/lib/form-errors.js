// Libraries
import React, { Component } from 'react'

class FormErrors extends Component {
  render () {
    const { errors } = this.props
    if (errors.length === 0) {
      return null
    }

    window.scrollTo(0, 0)
    return (
      errors.map((error, index) => {
        return <div key={index} className='c-account-errors'>
          <h1>There has been a problem processing your request:</h1>
          <li>{ error.detail }</li>
        </div>
      })
    )
  }
}

export default FormErrors
