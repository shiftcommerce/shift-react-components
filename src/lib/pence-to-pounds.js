/**
 * Convert the price from pence to pounds. For example:
 * 2547 pence becomes 25.47
 * @param  {Number} priceInPence
 * @return {Number} - The price
 */

export function penceToPounds (priceInPence) {
  return (priceInPence / 100).toFixed(2)
}
