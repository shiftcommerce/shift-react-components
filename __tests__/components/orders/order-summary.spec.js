// Libraries
import React from 'react'

// Component
import OrderSummary from '../../../src/components/orders/order-summary'

// Lib
import { decimalPrice } from '../../../src/lib/decimal-price'

// Fixtures
import order from '../../fixtures/order-confirmation'

test('renders correctly', () => {
  // arrange
  const lineItems = order.line_items

  // act
  const wrapper = mount(
    <OrderSummary order={order} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(lineItems[0].title)
  expect(wrapper).toIncludeText(lineItems[0].sku)
  expect(wrapper).toIncludeText(decimalPrice(order.total_inc_tax - order.shipping_total))
  expect(wrapper).toIncludeText(decimalPrice(order.shipping_total))
  expect(wrapper).toIncludeText(decimalPrice(order.total_inc_tax))
})
