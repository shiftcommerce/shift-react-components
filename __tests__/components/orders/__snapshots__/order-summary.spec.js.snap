// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<OrderSummary
  order={
    Object {
      "billing_address": Object {
        "attributes": Object {
          "address_line_1": "104 Quay One",
          "address_line_2": "",
          "city": "Leeds",
          "country": "GB",
          "first_name": "john",
          "last_name": "smith",
          "postcode": "LS9 8DS",
        },
        "id": "",
        "type": "addresses",
      },
      "cardToken": Object {
        "card": Object {
          "address_city": "Leeds",
          "address_country": "GB",
          "address_line1": "104 Quay One",
          "address_line1_check": "unchecked",
          "address_line2": "",
          "address_state": "Some state",
          "address_zip": "LS9 8DS",
          "address_zip_check": "unchecked",
          "brand": "Visa",
          "country": "US",
          "cvc_check": "unchecked",
          "dynamic_last4": null,
          "exp_month": 9,
          "exp_year": 2029,
          "funding": "credit",
          "id": "card_1DRhVLIZ9svPgtYNHTG1Vare",
          "last4": "4242",
          "metadata": Object {},
          "name": "john smith",
          "object": "card",
          "tokenization_method": null,
        },
        "client_ip": "90.198.102.132",
        "created": 1541083839,
        "id": "tok_1DRhVLIZ9svPgtYNDsRIAYyp",
        "livemode": false,
        "object": "token",
        "type": "card",
        "used": false,
      },
      "cardTokenRequested": false,
      "card_errors": false,
      "channel": "web",
      "coupons_applied": Array [],
      "currency": "GBP",
      "customer_account_id": null,
      "deallocated": false,
      "deallocated_at": null,
      "discount_summaries": Array [
        Object {
          "attributes": Object {
            "name": "",
            "promotion_id": 0,
            "total": 0,
          },
          "id": "",
          "type": "discount_summaries",
        },
      ],
      "email": "john.smith@shiftcommerce.com",
      "error": false,
      "free_shipping": false,
      "id": "95",
      "ip_address": "90.198.102.132",
      "line_items": Array [
        Object {
          "gift": false,
          "gift_unit_quantity": 0,
          "id": "127",
          "line_item_discounts": Array [],
          "sku": "1425689355217",
          "taxes": 0,
          "title": "SCS-04-CJ coffee jug 600ml",
          "tracking_url": null,
          "unit_price": 90.31,
          "unit_quantity": 1,
        },
        Object {
          "gift": false,
          "gift_unit_quantity": 0,
          "id": "128",
          "line_item_discounts": Array [],
          "sku": "bdm1",
          "taxes": 0,
          "title": "Brazil Daterra Masterpiece - Curupira",
          "tracking_url": null,
          "unit_price": 16.5,
          "unit_quantity": 1,
        },
      ],
      "loading": false,
      "paymentError": null,
      "placed_at": "2018-11-01T14:50:39.968Z",
      "reference": "_95_",
      "shipping_address": Object {
        "attributes": Object {
          "address_line_1": "228 Query Avenue",
          "address_line_2": "",
          "city": "Manchester",
          "country": "GB",
          "first_name": "Terry",
          "last_name": "Tibbs",
          "postcode": "M1 1TP",
        },
        "id": "",
        "type": "addresses",
      },
      "shipping_discount": 0,
      "shipping_discount_name": null,
      "shipping_method": Object {
        "attributes": Object {
          "created_at": "2018-11-01T14:50:35.544Z",
          "description": "Sunday 2nd December",
          "label": "Super Saver",
          "meta_attributes": Object {},
          "reference": "super_saver",
          "sku": "super_saver",
          "sub_total": 3.45,
          "tax": 2,
          "tax_rate": 20,
          "total": 3.45,
          "updated_at": "2018-11-01T14:50:35.544Z",
        },
        "id": "1",
        "type": "shipping_methods",
      },
      "shipping_total": 3.45,
      "status": "pending",
      "sub_total": 106.81,
      "tax": 0,
      "test": true,
      "total_inc_tax": 110.26,
    }
  }
>
  <div
    className="c-order__summary"
  >
    <div
      className="c-order__summary-header"
    >
      <div
        className="c-order__summary-header-title"
      >
        <h2>
          Your Order
        </h2>
      </div>
      <div
        className="c-order__summary-header-total"
      >
        <h2>
          £
          110.26
        </h2>
      </div>
    </div>
    <div
      className="c-order__summary-line-item"
      key="1425689355217"
    >
      <div
        className="c-order__summary-line-item-image"
      >
        <Image
          alt="SCS-04-CJ coffee jug 600ml"
          aria-label="SCS-04-CJ coffee jug 600ml"
          src="/static/placeholder.png"
        >
          <div>
            <picture>
              <img
                alt="SCS-04-CJ coffee jug 600ml"
                aria-label="SCS-04-CJ coffee jug 600ml"
                className="o-image"
                src="/static/placeholder.png"
                srcSet="/static/placeholder.png 769w"
              />
            </picture>
          </div>
        </Image>
      </div>
      <div
        className="c-order__summary-line-item-details"
      >
        <p
          className="c-order__summary-line-item-details-title u-bold"
        >
          SCS-04-CJ coffee jug 600ml
        </p>
        <p
          className="c-order__summary-line-item-details-sku"
        >
          1425689355217
        </p>
        <div
          className="c-order__summary-line-item-details-param"
          key="colour"
        >
          <div
            className="c-order__summary-line-item-details-param-key"
          >
            <p>
              colour
              :
            </p>
          </div>
          <div
            className="c-order__summary-line-item-details-param-value"
          >
            <p>
              Grey
            </p>
          </div>
        </div>
        <div
          className="c-order__summary-line-item-details-param"
          key="size"
        >
          <div
            className="c-order__summary-line-item-details-param-key"
          >
            <p>
              size
              :
            </p>
          </div>
          <div
            className="c-order__summary-line-item-details-param-value"
          >
            <p>
              L
            </p>
          </div>
        </div>
        <div
          className="c-order__summary-line-item-details-param"
          key="quantity"
        >
          <div
            className="c-order__summary-line-item-details-param-key"
          >
            <p>
              quantity
              :
            </p>
          </div>
          <div
            className="c-order__summary-line-item-details-param-value"
          >
            <p>
              1
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c-order__summary-line-item"
      key="bdm1"
    >
      <div
        className="c-order__summary-line-item-image"
      >
        <Image
          alt="Brazil Daterra Masterpiece - Curupira"
          aria-label="Brazil Daterra Masterpiece - Curupira"
          src="/static/placeholder.png"
        >
          <div>
            <picture>
              <img
                alt="Brazil Daterra Masterpiece - Curupira"
                aria-label="Brazil Daterra Masterpiece - Curupira"
                className="o-image"
                src="/static/placeholder.png"
                srcSet="/static/placeholder.png 769w"
              />
            </picture>
          </div>
        </Image>
      </div>
      <div
        className="c-order__summary-line-item-details"
      >
        <p
          className="c-order__summary-line-item-details-title u-bold"
        >
          Brazil Daterra Masterpiece - Curupira
        </p>
        <p
          className="c-order__summary-line-item-details-sku"
        >
          bdm1
        </p>
        <div
          className="c-order__summary-line-item-details-param"
          key="colour"
        >
          <div
            className="c-order__summary-line-item-details-param-key"
          >
            <p>
              colour
              :
            </p>
          </div>
          <div
            className="c-order__summary-line-item-details-param-value"
          >
            <p>
              Grey
            </p>
          </div>
        </div>
        <div
          className="c-order__summary-line-item-details-param"
          key="size"
        >
          <div
            className="c-order__summary-line-item-details-param-key"
          >
            <p>
              size
              :
            </p>
          </div>
          <div
            className="c-order__summary-line-item-details-param-value"
          >
            <p>
              L
            </p>
          </div>
        </div>
        <div
          className="c-order__summary-line-item-details-param"
          key="quantity"
        >
          <div
            className="c-order__summary-line-item-details-param-key"
          >
            <p>
              quantity
              :
            </p>
          </div>
          <div
            className="c-order__summary-line-item-details-param-value"
          >
            <p>
              1
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c-order__summary-totals"
    >
      <Sticker>
        <div
          aria-label="Order total summary"
          className="u-sticky"
        >
          <div>
            <dl
              aria-label="Order subtotal"
            >
              <dt>
                Total products:
              </dt>
              <dd>
                £
                106.81
              </dd>
            </dl>
            <dl
              aria-label="Order shipping costs"
            >
              <dt>
                Shipping costs:
              </dt>
              <dd>
                £
                3.45
              </dd>
            </dl>
            <dl
              aria-label="Order total"
              className="u-bold"
            >
              <dt>
                TOTAL:
              </dt>
              <dd>
                £
                110.26
              </dd>
            </dl>
            <dl>
              <dt>
                * Including VAT
              </dt>
            </dl>
          </div>
        </div>
      </Sticker>
    </div>
  </div>
</OrderSummary>
`;
